packages/react-avatar/src/components/Avatar/Avatar.scss

```diff
@@ -24,86 +14,65 @@
   display: inline-block;
   vertical-align: middle;
   flex-shrink: 0;
-  width: var(--avatarSize, var(--avatarSizeMedium));
-  height: var(--avatarSize, var(--avatarSizeMedium));
+  width: var(--avatar-size);
+  height: var(--avatar-size);
 }
 
 // Image slot
 .image {
   position: absolute;
-  border-radius: var(--avatarBorderRadius);
+  border-radius: var(--avatar-borderRadius);
   top: 0;
   left: 0;
-  width: var(--avatarSize, var(--avatarSizeMedium));
-  height: var(--avatarSize, var(--avatarSizeMedium));
-  height: 100%;
+  width: var(--avatar-size);
+  height: var(--avatar-size);
   object-fit: cover;
   vertical-align: top;
+  clip-path: var(--avatar-clipPath);
 }
 
 // Label slot
 .label {
   display: flex;
   align-items: center;
   justify-content: center;
-  background-color: var(--avatarFill);
+  background-color: var(--avatar-fill);
 
-  font-size: calc(var(--avatarSize) / 2.333);
-  width: var(--avatarSize, var(--avatarSizeMedium));
-  height: var(--avatarSize, var(--avatarSizeMedium));
+  font-size: var(--avatar-fontSize, calc(var(--avatar-size) / 2.333));
+  width: var(--avatar-size);
+  height: var(--avatar-size);
 
   vertical-align: top;
   text-align: center;
   padding: 0px;
   border-radius: 99999px;
+  clip-path: var(--avatar-clipPath);
 }
 
-// Status slot
-.status {
+// Badge slot
+.badge {
   position: absolute;
   bottom: 0;
   right: 0;
-  box-shadow: 0 0 0 var(--statusBorderWidth) var(--statusBorderColor);
-}
-
-// Root modifier when props.size_ smallest
-._size_smallest {
-  --avatarSize: var(--avatarSizeSmallest);
-}
-
-// Root modifier when props.size_ smaller
-._size_smaller {
-  --avatarSize: var(--avatarSizeSmaller);
-}
-
-// Root modifier when props.size_ small
-._size_small {
-  --avatarSize: var(--avatarSizeSmall);
-}
 
-// Root modifier when props.size_ large
-._size_large {
-  --avatarSize: var(--avatarSizeLarge);
-}
-
-// Root modifier when props.size_ larger
-// larger-abc123
-._size_larger {
-  --avatarSize: var(--avatarSizeLarger);
-}
-
-// Root modifier when props.size_ largest
-._size_largest {
-  --avatarSize: var(--avatarSizeLargest);
+  // Size the badge appropriately for this avatar size
+  --badge-size: calc(var(--avatar-size) / 4);
 }
 
 // Modifiers
 ._square {
   .image {
-    border-radius: var(--squareAvatarBorderRadius);
+    border-radius: var(--avatar-square-borderRadius, calc(var(--avatar-size) / 12));
```

Comment:
We probably should rework this a little: at first I thought `avatar-square-borderRadius` was correct, but I don't think this approach scales. What about when you want square to also change the border width? or the border color? Or maybe square needs box shadow.

Instead, we should look at class modifiers as a different "variant" of the control, where a variant simply defines new set of token values:

```css
._square .image {
  --avatar-borderRadius: calc(var(--avatar-size) / 12);
}
```